### ISSUE TITLE
Simple HowTo?

### ISSUE THREAD

**Issue by vitasam** [suutari/waterwatch#6]
Hello,
quite interesting project. 
Do you have any kind of How-to - how to tune and calibrate `meterelf` for particular water meter images?

**Comment by suutari** [comment]
Hi! And thanks!

TL;DR; It's currently very much manual work involving running code snippets in Python shell and there's a chance that the code is not generic enough to handle the layout of your meter without little changes.

There's not any how-to for the calibration and I'm not very certain how well it will work with other kind of meters. It could be that you need to do some small code changes too, even though I have tried to make it configurable and to allow it to be used with any kind of meters with just different parameters in the YAML file.

The calibration values I have generated by taking multiple (e.g. 100--500) sample images of the meter from the same fixed position and light conditions as will be used for the reading eventually. Then two files need to be generated with help of some Python code and these images:
1. Grayscale presentation of the area where the dials are. It looks like this:
    ![example grayscale dials image](https://github.com/suutari/meterelf/raw/722cc580a4ca0dc88b0117e984a6be6226ab28c1/sample-images1/dials_gray.png)
2. A parameters file with some preliminary values. If I remember correctly, I've generated the values with help of GIMP and some python code. The final values for the center coordinates and diameters can be calibrated with `find_dial_centers` function from `_calibration.py`. Here's an example of the parameters file: https://github.com/suutari/meterelf/blob/722cc580a4ca0dc88b0117e984a6be6226ab28c1/sample-images1/params.yml

The grayscale image I generated by combining the sample images with the `get_average_meter_image` function (from `_calibration.py`) and the cropping it with GIMP.

There's some sample code on how to calibrate the dial center values in the tests:
https://github.com/suutari/meterelf/blob/722cc580a4ca0dc88b0117e984a6be6226ab28c1/tests/test_meterelf.py#L118-L136

I'm afraid that these instructions are not very clear and probably I've forgotten quite many steps too, because it has been a long time since I've worked on this last time. You would probably have to dig deeper into code to learn the details. If you do and write better instructions in the process, I would be more than happy to accept a PR for the documentation or code improvements.

**Comment by vitasam** [comment]
Hi, thanks for the info!
I have pretty much the same type of water meter, with 4 dials. I’m planning to use Raspberry Pi with camera V2. Furthermore, I thought to apply something based on neural network (e.g. this [project](https://github.com/jomjol/water-meter-system-complete)), but decided to try this OpenCV approach. 

I will try to calibrate. What do you think, does it make sense to make calibration procedure as `meterelf` standard feature (API) rather than the bunch of separate tests?

**Comment by suutari** [comment]
> I will try to calibrate. What do you think, does it make sense to make calibration procedure as *meterelf* standard feature (API) rather than the bunch of separate tests?

Yes, definitely. Only reason that it doesn't exist yet is lack of time. I managed to get it working for my case without a proper API for calibration and just haven't put so much effort into this after it worked for me.

**Comment by vitasam** [comment]
Great!

Just anotjer question (I never been working with library sources before) - what is a best approach to work with your code base under Windows? Do I need to install the lib with pip install meterelf or I can import functionality to test from the folder itself?

**Comment by suutari** [comment]
> Just anotjer question (I never been working with library sources before) - what is a best approach to work with your code base under Windows? Do I need to install the lib with pip install meterelf or I can import functionality to test from the folder itself?

You shouldn't install meterelf from PyPI, but instead clone the code base and have a virtualenv for it. Install the required dependencies to the virtualenv with pip. Ensure that the virtualenv is active and then you should be able to install the deps with `pip install -r requirements.txt` command. It's also possible that one of those [requirements](https://github.com/suutari/meterelf/blob/master/requirements.txt) doesn't work for Windows. In that case you could try a different version.

To install meterelf itself to the virtualenv, use `pip install -e .` when you're inside the checkout directory (and virtualenv is activated). This should add the `meterelf` script to the PATH and then you should be able to run it with that name.  Another option is to run it with `python -m meterelf`

**Comment by vitasam** [comment]
Thanks! I got it working under Windows 10 with Miniconda, I've found it quite handy to work with different Python versions and environments.

I just created a pull request with updated README file and conda-specific requirements.txt file, in order not to forget how to get the stuff working.
